project/code/application.py

```diff
@@ -0,0 +1,60 @@
+import argparse
+from keychain import Storage
+
+
+def main(arguments):
+    storage = allocate_application(arguments)
+
+    # Adding a key-value pair to the storage.
+    key = "info8002"
+    value = "fun"
+    callback = storage.put(key, value, block=False)
+
+    # Depending on how fast your blockchain is,
+    # this will return a proper result.
+    print(storage.retrieve(key))
+
+    # Using the callback object,
+    # you can also wait for the operation to be completed.
+    callback.wait()
+
+    # Now the key should be available,
+    # unless a different node `put` a new value.
+    print(storage.retrieve(key))
+
+    # Show all values of the key.
+    print(storage.retrieve_all(key))
+
+    # XXX: not defined!
+    application.terminate()
```

Comment:
this is not defined. shall we remove it?

Reply:
Yes, forgot to remove this. Thanks.